extension/source/Home/Wallet/Wallets/Recovery/StepThreeRecover.tsx

```diff
@@ -0,0 +1,84 @@
+import { ethers } from "ethers";
+import { Download } from "phosphor-react";
+import { FunctionComponent, useState } from "react";
+import Button from "../../../../components/Button";
+import { useQuill } from "../../../../QuillContext";
+
+const StepThreeRecover: FunctionComponent<{
+  onComplete: () => void;
+  walletAddress: string;
+}> = ({ onComplete, walletAddress }) => {
+  const { rpc } = useQuill();
+
+  const [salt, setSalt] = useState<string>("");
+  const [instantWalletAddress, setInstantWalletAddress] = useState<string>("");
+
+  const handleRecover = async () => {
+    const pk = await rpc.lookupPrivateKey(walletAddress);
+    await rpc.addRecoveryWallet(
+      instantWalletAddress,
+      ethers.utils.hexlify(ethers.utils.toUtf8Bytes(salt)),
```

Comment:
is this the right way?

Reply:
Just tested this out and I think that produces a string that's the wrong length.

I switched it to this and tested it out locally and it seemed to work
```
ethers.utils.formatBytes32String(salt)
```

Reply:
gotcha! I was confused too. updated in da2264c